File: Connect-Objective-CSampleApp/ConnectObjective-CSampleApp/ResultViewController.m
//
//  ResultViewController.m
//  ConnectObjective-CSampleApp
//
//  Created by Tapas Behera on 8/8/17.
//  Copyright © 2017 Acuant. All rights reserved.
//

#import "ResultViewController.h"
#import "UIImageView+LoadImage.h"

typedef enum AssureIDResult
{
    Unknown = 0,
    Passed = 1,
    Failed = 2,
    Skipped = 3,
    Caution = 4,
    Attention = 5

}AssureIDResult;

@interface ResultViewController ()

@property(nonatomic,strong) IBOutlet UIImageView* frontCardImageView;
@property (strong, nonatomic) IBOutlet UIImageView *backCardImageView;
@property(nonatomic,strong) IBOutlet UIImageView* faceImageView;
@property (strong, nonatomic) IBOutlet UIImageView *signImageView;
@property (strong, nonatomic) IBOutlet UIButton *backButton;
@property (strong, nonatomic) IBOutlet UITextView *resultTextView;

@end

@implementation ResultViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    _separator = @"-";
    [self initUI];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)backTapped:(id)sender {
    [self dismissViewControllerAnimated:YES completion:nil];
    [_cancelDelegate didFinishShowingResult];
}

-(void)initUI{
    NSString* resultStr = @"";
    NSString* faceImageURL=@"";
    NSString* signImageURL=@"";
    NSArray* alerts = [_cardData objectForKey:@"Alerts"];
    int Result = [[_cardData objectForKey:@"Result"] intValue] ;
    NSArray* images = [_cardData objectForKey:@"Images"];
    NSArray* fields = [_cardData objectForKey:@"Fields"];
    if(Result==Passed || Result==Failed || Result==Unknown){
        if(Result==Passed){
            resultStr =  [[[@"Authentication Result" stringByAppendingString:_separator] stringByAppendingString:@"Passed"] stringByAppendingString:@"\n"];
        }else if(Result==Failed){
            resultStr =  [[[@"Authentication Result" stringByAppendingString:_separator] stringByAppendingString:@"Failed"] stringByAppendingString:@"\n"];
        }else if(Result==Unknown){
            resultStr =  [[[@"Authentication Result" stringByAppendingString:_separator] stringByAppendingString:@"Unknown"] stringByAppendingString:@"\n"];
        }
    }else{
        for(NSDictionary* alert in alerts){
            if([[alert objectForKey:@"Result"] intValue]==Result){
                resultStr =  [resultStr stringByAppendingString:[[[@"Alert" stringByAppendingString:_separator] stringByAppendingString:[alert objectForKey:@"Key"]] stringByAppendingString:@"\n"]];
            }
        }
    }
    
    for(NSDictionary* field in fields){
        NSString* f = [NSString stringWithFormat:@"%@%@%@",[field objectForKey:@"Name"],_separator,[field objectForKey:@"Value"]];
        if([[field objectForKey:@"Name"] isEqualToString:@"Photo"]){
            faceImageURL = [field objectForKey:@"Value"];
        }else if([[field objectForKey:@"Name"] isEqualToString:@"Signature"]){
            signImageURL = [field objectForKey:@"Value"];
        }else{
            resultStr =  [[resultStr stringByAppendingString:f] stringByAppendingString:@"\n"];
        }
        
        
    }
    if(![faceImageURL isEqualToString:@""]){
        [_faceImageView downloadedFromurlStr:faceImageURL username:_username password:_password];
    }
    if(![signImageURL isEqualToString:@""]){
        [_signImageView downloadedFromurlStr:signImageURL username:_username password:_password];
    }
    
    for(NSDictionary* imageDict in images){
        if([imageDict objectForKey:@"Side"]){
            int side = [[imageDict objectForKey:@"Side"] intValue];
            if(side==0 && [imageDict objectForKey:@"Uri"]){
                [_frontCardImageView downloadedFromurlStr:[imageDict objectForKey:@"Uri"] username:_username password:_password];
            }else if(side==1 && [imageDict objectForKey:@"Uri"]){
                [_backCardImageView downloadedFromurlStr:[imageDict objectForKey:@"Uri"] username:_username password:_password];
            }
        }
    }
    _resultTextView.text=resultStr;
}


@end

File: Sample-Connect-Swift-App/AcuantiOSSDKAssureIDSwiftSample/ResultViewController.swift
//
//  ResultViewController.swift
//  AcuantiOSSDKSwiftSample
//
//  Created by Tapas Behera on 7/28/17.
//  Copyright © 2017 Acuant. All rights reserved.
//

import UIKit

enum AssureIDResult : Int{
    case Unknown = 0;
    case Passed = 1;
    case Failed = 2;
    case Skipped = 3;
    case Caution = 4;
    case Attention = 5;
}


class ResultViewController: UIViewController{
    
    var cancelDelegate:ResultCancelDelegate!
    var cardData:NSDictionary!
    var cardType:AcuantCardType!
    var separator: String!
    var username:String!
    var password:String!
    
    @IBOutlet var frontImage: UIImageView!
    @IBOutlet var backImage: UIImageView!
    @IBOutlet var faceImage: UIImageView!
    @IBOutlet var signImage: UIImageView!
    
    @IBOutlet var resultTextView: UITextView!
    @IBAction func backTapped(_ sender: Any) {
        self.dismiss(animated: true, completion:{
            self.cancelDelegate.didFinishShowingResult()
        })
        
    }
    override func viewDidLoad() {
        super.viewDidLoad()
        separator = "  -  "
        initializeUI();
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func initializeUI(){
        var fieldString : String = "";
        
        let Result : Int = cardData["Result"] as! Int
        
        let Alerts  = cardData["Alerts"]
        
        if(Result==0 || Result==1 || Result==2){
            if(Result==0){
                fieldString = "Authentication Result : Unknown\n"
            }else if(Result==1){
                fieldString = "Authentication Result : Passed\n"
            }else if(Result==2){
                fieldString = "Authentication Result : Failed\n"
            }else{
                for alert in Alerts as! [[String:String]]{
                    fieldString.append("Alert"+separator+alert["Key"]!)
                }
            }
        }
        
        
        let fieldArray = cardData["Fields"]
        
        
        var faceImageURL : String = "";
        var signatureImageURL : String = "";
        
        for field in fieldArray as! [NSDictionary] {
            let key : String = field["Name"]! as! String
            let value : String = field["Value"]! as! String
            if(key=="Photo"){
                faceImageURL = value
            }else if(key=="Signature"){
                signatureImageURL = value
            }else{
                var str : String =   key + separator + value
                str.append("\n")
                fieldString.append(str);
            }
        }
        
        
        resultTextView.text = fieldString
        if(faceImageURL != ""){
            faceImage.downloadedFrom(urlStr: faceImageURL, username: username, password: password)
        }
        
        if(signatureImageURL != ""){
            signImage.downloadedFrom(urlStr: signatureImageURL, username: username, password: password)
        }
        
        
        let images = cardData["Images"]
        for imageDict in images as! [NSDictionary]{
            if(imageDict["Side"] != nil){
                let side : Int = imageDict["Side"] as! Int;
                if(side == 0 && imageDict["Uri"] != nil){
                    frontImage.downloadedFrom(urlStr: imageDict["Uri"]! as! String, username: username, password: password)
                }else if(side == 1 && imageDict["Uri"] != nil){
                    backImage.downloadedFrom(urlStr: imageDict["Uri"]! as! String, username: username, password: password)
                }
            }
        }
        
    }
}
